build-tools/multiplatform/src/main/kotlin/aditogether/buildtools/multiplatform/MultiPlatformPlugin.kt

```diff
@@ -0,0 +1,37 @@
+package aditogether.buildtools.multiplatform
+
+import aditogether.buildtools.jvm.JvmOptions
+import aditogether.buildtools.utils.apply
+import aditogether.buildtools.utils.configure
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinMultiplatformPluginWrapper
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinJvmTargetPreset
+import org.jetbrains.kotlin.gradle.targets.jvm.KotlinJvmTarget
+
+/**
+ * Applies a shared Kotlin multi-platform configuration to the given project.
+ * This plugin supports the following compilation targets:
+ * - JVM
+ */
+@Suppress("unused")
+internal class MultiPlatformPlugin : Plugin<Project> {
+    override fun apply(target: Project) {
+        target.pluginManager.apply<KotlinMultiplatformPluginWrapper>()
+
+        target.extensions.configure<KotlinMultiplatformExtension> { ext ->
+            ext.targetFromPreset(KotlinJvmTargetPreset(target), ::configureJvmTarget)
+        }
+    }
+
+    private fun configureJvmTarget(target: KotlinJvmTarget) {
+        target.compilations.all { compilation ->
+            compilation.compilerOptions.configure {
+                allWarningsAsErrors.set(JvmOptions.WARNINGS_AS_ERRORS)
+                jvmTarget.set(JvmTarget.fromTarget(JvmOptions.JAVA_VERSION.toString()))
```

Comment:
Not really relevant at this point, but we should consider that in future we may not want apply the same targets to all the KMP modules. An example could be the Android target: we want to apply it only where needed, like where we need the Driver for SqlDelight, but for the majority only jvm would be enough.

Reply:
Good point. I think we have 2 solutions:

1. configure already existing compilation targets applying our shared config on top of the user-specific one
2. provide an extension to configure this plugin defining which targets should configured

Solution 1. is easier to introduce, easier to maintain for build-tools and more flexible. Solution 2. instead, promotes strictness (everything will be controlled by build-tools).

I'd go with 1 because in the first period we'll have a lot of changes probably and maintaining 2. can be an hassle.

To be clear what I mean with solution 1.:

```groovy
// This will add a shared config to the targets below.
apply plugin: 'aditogether.multiplatform'

kotlin {
  jvm()
}
```

What do you think?

Reply:
Actually, I think this discussion applies only to the Android target, so maybe we can define here all the shared targets and let the user define the Android one, in the module where it's needed. I think this shouldn't be a problem, right?
If we will even need something more fine grained, we can evaluate different solutions

Reply:
Honestly I'd find confusing having some targets created inside this plugin and some outside. I would prefer them all inside (enabled/disable from the outside) or all outside. I wouldn't handle Android as an exception

Reply:
Makes sense! So at this point I'd keep everything here in our plugin and evaluate a different solution when/if needed

Reply:
Maybe I've misunderstood, are you suggesting to go for solution 1. or 2.? Or another one?

Reply:
I meant we can keep it as is in your code for now

Reply:
Ok, fair enough!